opencog/atoms/core/Context.cc

```diff
@@ -81,7 +81,7 @@ bool Context::is_free_variable(const Handle& h) const
 	Type t = h->get_type();
 	return (t == VARIABLE_NODE or t == GLOB_NODE)
 		and quotation.is_unquoted()
-		and not contains(shadow, h);
+		and not content_contains(HandleSeq(shadow.begin(), shadow.end()), h);
```

Comment:
Actually, this change should not be necessary.  That is because `opencog::contains` is specialized for sets, and `Handle::operator<` is already based on content.  Now if the compiler somehow chooses the generic `opencog::contains` instead of the specialized one for sets, then it would indeed introduce a bug that this change would fix.

@Habush, does the absence of that change breaks something on your end?  Or did you fix it solely based on code review?

 Of course the alternative is to have `Handle::operator==` being based on content, as I have repeated many times (maybe I'll just make it happen one day, the last time I tried it broke a bunch of utests, and so I gave up).

Reply:
> `Handle::operator==` ...  broke a bunch of utests

That's because the utests do some mighty weird things, comparing atoms that are in and not in atomspaces, deleting them, comparing deleted atoms, you name it, all-around craziness. They're also quite gnarly. I don't want to invest time in "fixing" them, and I think removing them would be a bad idea. Other than that, yes, I think I now finally agree that `Handle::operator==` could be overloaded. (It should do a pointer compare first, and then mark the rest of the body with `unlikely` so that the optimizer can do thee right thing.

Reply:
`unlikely`?  Sick (as in cool) https://en.cppreference.com/w/cpp/language/attributes/likely

Anyway, changing `Handle::operator==` to be content based is not urgent on my end either.